### PR TITLE
ci: enable wasm lint/build/test on every PR and push to main.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

not running these could lead to subtle errors passing through with changes to ibc-go. Since we consume ibc-go as a library, it is a good idea to run these tests whenever it changes.

closes: #XXXX

### Commit Message / Changelog Entry

```text
ci: enable wasm lint/build/test on every PR and push to main.
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by crodriguezvega — APPROVED** [review]
💯

**Review by damiannolan — APPROVED** [review]
I think we originally had this under stricter rules because the contract tests did not have the mock vm yet. 

Thanks @DimitrisJim 👍🏻

File: .github/workflows/wasm-client.yml [hunk 1]
```diff
@@ -2,10 +2,9 @@ name: Wasm Light-Client
 # This workflow runs when a PR is opened that targets code that is part of the wasm light-client.
 on:
   pull_request:
-    paths:
-      - '.github/workflows/wasm-client.yml'
-      - 'modules/light-clients/08-wasm/**'
-      - 'proto/ibc/lightclients/wasm/**'
+  push:
+    branches:
+      main
```
Comment (chatton) [inline .github/workflows/wasm-client.yml:7]: do we need to run them again on merge to main? The code on the PR should already be synced with main when they're running on the PR right?

**Review by chatton — APPROVED** [review]
LGTM, left once comment about (possibly) a redundant test run

**Pull request merged**